File: src/jsMain/kotlin/kingdom/lite/ui/turn/ResourcesPhase.kt
package kingdom.lite.ui.turn

import kingdom.lite.model.*
import kingdom.lite.ui.styles.ResourceStyles

/**
 * Resources Phase content for the Kingdom Sheet
 * Handles resource collection, consumption, military support, and build queue
 */
class ResourcesPhase(
    private val kingdomState: KingdomState,
    private val turnManager: TurnManager
) {
    
    private val production = kingdomState.calculateProduction()
    
    fun render(): String = buildString {
        // Include resource styles
        append(ResourceStyles.render())
        
        append("""
            <div class="resources-phase-container">
                <!-- Part 1: Resource Display at the Top -->
                ${renderResourceDisplay()}
                
                <!-- Part 2: Resource Steps Underneath -->
                ${renderResourceSteps()}
            </div>
        """)
    }
    
    /**
     * Part 1: Render the resource display at the top
     */
    private fun renderResourceDisplay(): String = renderResourceDashboard()
    
    /**
     * Part 2: Render all resource steps in a scrollable container
     */
    private fun renderResourceSteps(): String = buildString {
        append("""
            <div class="resource-steps-scroll">
                ${renderStep1CollectResources()}
                ${renderStep2FoodConsumption()}
                ${renderStep3MilitarySupport()}
                ${renderStep4BuildQueue()}
            </div>
        """)
    }
    
    /**
     * Render the resource dashboard (non-sticky)
     */
    private fun renderResourceDashboard(): String = buildString {
        append("""
            <div class="resource-dashboard-wrapper">
                <div class="resource-dashboard">
                    ${renderCompactResourceCard("food", kingdomState.resources["food"] ?: 0)}
                    ${renderCompactResourceCard("lumber", kingdomState.resources["lumber"] ?: 0)}
                    ${renderCompactResourceCard("stone", kingdomState.resources["stone"] ?: 0)}
                    ${renderCompactResourceCard("ore", kingdomState.resources["ore"] ?: 0)}
                    ${renderCompactResourceCard("gold", kingdomState.resources["gold"] ?: 0)}
                </div>
            </div>
        """)
    }
    
    /**
     * Render a compact resource card for the sticky dashboard
     */
    private fun renderCompactResourceCard(resource: String, value: Int): String {
        val (icon, color) = when (resource) {
            "food" -> "fa-wheat-awn" to "#8B4513"
            "lumber" -> "fa-tree" to "#228B22"
            "stone" -> "fa-cube" to "#708090"
            "ore" -> "fa-mountain" to "#4B0082"
            "gold" -> "fa-coins" to "#FFD700"
            else -> "fa-question" to "#666666"
        }
        
        return """
            <div class="resource-card">
                <i class="fas $icon resource-icon" style="color: $color;"></i>
                <div class="resource-value">$value</div>
                <div class="resource-label">${resource.capitalize()}</div>
            </div>
        """
    }
    
    /**
     * Step 1: Collect Resources
     */
    private fun renderStep1CollectResources(): String = buildString {
        val isCompleted = kingdomState.isPhaseStepCompleted("resources_collect")
        
        append("""
            <div class="phase-step-container" style="position: relative;">
                ${if (isCompleted) """<i class="fas fa-check-circle phase-step-complete"></i>""" else ""}
                <strong>Step 1: Collect Resources and Revenue</strong>
                
                <!-- Always show production summary at the top -->
                ${if (production.isNotEmpty() || kingdomState.hexes.any { it.worksite != null }) {
                    """
                    <div class="production-summary" style="margin: 15px 0; padding: 15px; background: linear-gradient(135deg, #f5f5f5 0%, #e5e5e5 100%); border-radius: 8px; border: 1px solid #d1d1d1;">
                        <div style="display: flex; justify-content: space-between; align-items: center; margin-bottom: 10px;">
                            <div class="production-summary-title" style="font-weight: bold; font-size: 16px;">Expected Income This Turn:</div>
                            <div style="font-weight: bold; color: #22c55e;">
                                ${if (production.isNotEmpty()) {
                                    production.entries.joinToString(" | ") { "+${it.value} ${it.key.capitalize()}" }
                                } else {
                                    "No Production"
                                }}
                            </div>
                        </div>
                        
                        ${if (kingdomState.hexes.any { it.worksite != null }) {
                            """
                            <details style="margin-top: 10px;">
                                <summary style="cursor: pointer; color: #666; font-size: 14px;">View Worksite Details</summary>
                                <ul class="worksite-list" style="margin-top: 10px;">
                                    ${kingdomState.hexes.filter { it.worksite != null }.joinToString("") { hex ->
                                        val prod = hex.getProduction()
                                        """<li class="worksite-item">
                                            <span>${hex.name ?: "Hex ${hex.id}"} (${hex.terrain})</span>
                                            <span>${if (prod.isNotEmpty()) {
                                                prod.entries.joinToString(", ") { "${it.value} ${it.key.capitalize()}" }
                                            } else {
                                                "No production"
                                            }}</span>
                                        </li>"""
                                    }}
                                </ul>
                            </details>
                            """
                        } else {
                            ""
                        }}
                    </div>
                    """
                } else {
                    """<div style="margin: 15px 0; padding: 15px; background: #f9f9f9; border-radius: 8px; border: 1px solid #e5e5e5; text-align: center; color: #666; font-style: italic;">
                        No worksites currently producing resources
                    </div>"""
                }}
                
                ${if (isCompleted) {
                    """<div style="margin-top: 10px; color: #666; font-style: italic; text-align: center;">Resources have been collected for this turn.</div>"""
                } else {
                    ""
                }}
                
                ${if (!isCompleted) {
                    """<button class="turn-action-button" onclick="window.executeResourceStep1()">
                        <i class="fas fa-hand-holding-usd"></i> Collect Resources
                    </button>"""
                } else {
                    """<button class="turn-action-button" disabled style="opacity: 0.5;">
                        <i class="fas fa-check"></i> Resources Collected
                    </button>"""
                }}
            </div>
        """)
    }
    
    /**
     * Step 2: Food Consumption
     */
    private fun renderStep2FoodConsumption(): String = buildString {
        val isCompleted = kingdomState.isPhaseStepCompleted("resources_consumption")
        val totalFoodNeeded = kingdomState.getTotalFoodConsumption()
        val currentFood = kingdomState.resources["food"] ?: 0
        val shortage = maxOf(0, totalFoodNeeded - currentFood)
        
        append("""
            <div class="phase-step-container" style="position: relative;">
                ${if (isCompleted) """<i class="fas fa-check-circle phase-step-complete"></i>""" else ""}
                <strong>Step 2: Food Consumption</strong>
                
                <div style="margin: 10px 0;">
                    <div style="display: flex; justify-content: space-around; flex-wrap: wrap; gap: 20px;">
                        <div style="text-align: center;">
                            <i class="fas fa-home" style="font-size: 32px; color: #8B4513;"></i>
                            <div style="font-size: 20px; font-weight: bold; margin: 2px 0;">
                                ${kingdomState.settlements.sumOf { it.foodConsumption }}
                            </div>
                            <div style="font-size: 11px; color: #666;">Settlement Consumption</div>
                        </div>
                        
                        <div style="text-align: center;">
                            <i class="fas fa-shield-alt" style="font-size: 32px; color: #4682B4;"></i>
                            <div style="font-size: 20px; font-weight: bold; margin: 2px 0;">
                                ${kingdomState.armies.size}
                            </div>
                            <div style="font-size: 11px; color: #666;">Army Consumption</div>
                        </div>
                        
                        <div style="text-align: center;">
                            <i class="fas fa-wheat-awn" style="font-size: 32px; color: ${if (shortage > 0) "#ef4444" else "#22c55e"};"></i>
                            <div style="font-size: 20px; font-weight: bold; margin: 2px 0; color: ${if (shortage > 0) "#ef4444" else "#22c55e"};">
                                $currentFood / $totalFoodNeeded
                            </div>
                            <div style="font-size: 11px; color: #666;">Available / Required</div>
                        </div>
                    </div>
                    
                    ${if (shortage > 0 && !isCompleted) {
                        """<div class="consumption-warning">
                            <i class="fas fa-exclamation-triangle"></i>
                            <strong>Warning:</strong> Food shortage of $shortage will cause +$shortage Unrest!
                        </div>"""
                    } else {
                        ""
                    }}
                </div>
                
                ${if (!isCompleted) {
                    """<button class="turn-action-button" onclick="window.executeResourceStep2()">
                        <i class="fas fa-utensils"></i> Pay Food Consumption
                    </button>"""
                } else {
                    """<button class="turn-action-button" disabled style="opacity: 0.5;">
                        <i class="fas fa-check"></i> Consumption Paid
                    </button>"""
                }}
            </div>
        """)
    }
    
    /**
     * Step 3: Military Support
     */
    private fun renderStep3MilitarySupport(): String = buildString {
        val isCompleted = kingdomState.isPhaseStepCompleted("resources_military")
        val totalSupport = kingdomState.getTotalArmySupport()
        val armyCount = kingdomState.armies.size
        val unsupportedCount = maxOf(0, armyCount - totalSupport)
        
        append("""
            <div class="phase-step-container" style="position: relative;">
                ${if (isCompleted) """<i class="fas fa-check-circle phase-step-complete"></i>""" else ""}
                <strong>Step 3: Military Support</strong>
                
                <div class="army-support-display">
                    <div class="support-status ${when {
                        unsupportedCount > 0 -> "danger"
                        armyCount == totalSupport -> "warning"
                        else -> "good"
                    }}">
                        <i class="fas fa-shield-alt army-icon"></i>
                        <div>
                            <div style="font-size: 18px; font-weight: bold;">$armyCount / $totalSupport</div>
                            <div style="font-size: 12px; color: #666;">Armies / Capacity</div>
                        </div>
                    </div>
                    
                    ${if (unsupportedCount > 0) {
                        """<div class="support-status danger">
                            <i class="fas fa-exclamation-triangle" style="font-size: 32px; color: #ef4444;"></i>
                            <div>
                                <div style="font-size: 18px; font-weight: bold; color: #ef4444;">$unsupportedCount</div>
                                <div style="font-size: 12px; color: #666;">Unsupported</div>
                            </div>
                        </div>"""
                    } else {
                        ""
                    }}
                </div>
                
                ${if (unsupportedCount > 0 && !isCompleted) {
                    """<div class="consumption-warning">
                        <i class="fas fa-exclamation-triangle"></i>
                        <strong>Warning:</strong> $unsupportedCount unsupported ${if (unsupportedCount == 1) "army" else "armies"} will require morale checks!
                    </div>"""
                } else if (armyCount == 0) {
                    """<div style="margin-top: 10px; color: #666; font-style: italic; text-align: center;">No armies currently fielded</div>"""
                } else {
                    ""
                }}
                
                ${if (!isCompleted) {
                    """<button class="turn-action-button" onclick="window.executeResourceStep3()">
                        <i class="fas fa-flag"></i> Process Military Support
                    </button>"""
                } else {
                    """<button class="turn-action-button" disabled style="opacity: 0.5;">
                        <i class="fas fa-check"></i> Support Processed
                    </button>"""
                }}
            </div>
        """)
    }
    
    /**
     * Step 4: Build Queue
     */
    private fun renderStep4BuildQueue(): String = buildString {
        val isCompleted = kingdomState.isPhaseStepCompleted("resources_build")
        
        append("""
            <div class="phase-step-container" style="position: relative;">
                ${if (isCompleted) """<i class="fas fa-check-circle phase-step-complete"></i>""" else ""}
                <strong>Step 4: Process Build Queue</strong>
                
                ${if (kingdomState.buildQueue.isNotEmpty()) {
                    """<div class="build-queue-container">
                        <div style="margin-bottom: 15px; font-weight: bold;">
                            Available Resources: 
                            ${listOf("lumber", "stone", "ore").joinToString(", ") { 
                                "${kingdomState.resources[it] ?: 0} ${it.capitalize()}"
                            }}
                        </div>
                        
                        ${kingdomState.buildQueue.joinToString("") { project ->
                            renderBuildProject(project, isCompleted)
                        }}
                    </div>"""
                } else {
                    """<div style="margin-top: 10px; color: #666; font-style: italic; text-align: center;">No construction projects in queue</div>"""
                }}
                
                ${if (!isCompleted && kingdomState.buildQueue.isNotEmpty()) {
                    """<button class="turn-action-button" onclick="window.executeResourceStep4()">
                        <i class="fas fa-hammer"></i> Apply to Construction
                    </button>"""
                } else if (isCompleted) {
                    """<button class="turn-action-button" disabled style="opacity: 0.5;">
                        <i class="fas fa-check"></i> Resources Applied
                    </button>"""
                } else {
                    ""
                }}
            </div>
        """)
    }
    
    /**
     * Render a single resource card
     */
    private fun renderResourceCard(resource: String, amount: Int, showChange: Boolean): String {
        val (icon, color) = when (resource) {
            "food" -> "fa-wheat-awn" to "#8B4513"
            "lumber" -> "fa-tree" to "#228B22"
            "stone" -> "fa-cube" to "#708090"
            "ore" -> "fa-mountain" to "#4B0082"
            "gold" -> "fa-coins" to "#FFD700"
            else -> "fa-question" to "#666666"
        }
        
        val currentValue = kingdomState.resources[resource] ?: 0
        
        return """
            <div class="resource-card">
                <i class="fas $icon resource-icon" style="color: $color;"></i>
                <div class="resource-value">${if (showChange) currentValue else currentValue}</div>
                <div class="resource-label">${resource.capitalize()}</div>
                ${if (showChange && amount > 0) {
                    """<div class="resource-change positive">+$amount</div>"""
                } else {
                    ""
                }}
            </div>
        """
    }
    
    /**
     * Render a build project card
     */
    private fun renderBuildProject(project: BuildProject, isCompleted: Boolean): String {
        val percentage = project.getCompletionPercentage()
        val remaining = project.getRemainingCost()
        
        return """
            <div class="build-project-card">
                <div class="build-project-header">
                    <span class="build-project-name">${project.structureName}</span>
                    <span class="build-project-tier">Tier ${project.tier}</span>
                </div>
                
                <div class="progress-bar-container">
                    <div class="progress-bar-fill" style="width: $percentage%;">
                        <span class="progress-bar-text">$percentage%</span>
                    </div>
                </div>
                
                <div style="font-size: 14px; color: #666; margin-top: 5px;">
                    ${if (remaining.isNotEmpty()) {
                        "Needs: ${remaining.entries.joinToString(", ") { "${it.value} ${it.key}" }}"
                    } else {
                        "Complete!"
                    }}
                </div>
                
                ${if (!isCompleted && remaining.isNotEmpty()) {
                    """<div class="resource-allocation">
                        ${remaining.entries.joinToString("") { (resource, _) ->
                            val available = kingdomState.resources[resource] ?: 0
                            val pending = project.pendingAllocation[resource] ?: 0
                            """
                            <div class="allocation-control">
                                <i class="fas ${getResourceIcon(resource)}" style="color: ${getResourceColor(resource)};"></i>
                                <button class="allocation-button" onclick="window.adjustAllocation('${project.id}', '$resource', -1)" 
                                    ${if (pending <= 0) "disabled" else ""}>−</button>
                                <span class="allocation-value">$pending</span>
                                <button class="allocation-button" onclick="window.adjustAllocation('${project.id}', '$resource', 1)"
                                    ${if (pending >= available) "disabled" else ""}>+</button>
                            </div>
                            """
                        }}
                    </div>"""
                } else {
                    ""
                }}
            </div>
        """
    }
    
    private fun getResourceIcon(resource: String): String = when (resource) {
        "lumber" -> "fa-tree"
        "stone" -> "fa-cube"
        "ore" -> "fa-mountain"
        else -> "fa-question"
    }
    
    private fun getResourceColor(resource: String): String = when (resource) {
        "lumber" -> "#228B22"
        "stone" -> "#708090"
        "ore" -> "#4B0082"
        else -> "#666666"
    }
}

File: src/ui/turn/ResourcesPhase.ts
// Resources Phase UI component for PF2e Kingdom Lite
// Auto-converted and fixed from ResourcesPhase.kt

import { html } from '../html-helpers';
import { KingdomState, BuildProject } from '../../models/KingdomState';
import { TurnManager } from '../../models/TurnManager';
import { ResourceStyles } from '../styles/ResourceStyles';

/**
 * Resources Phase content for the Kingdom Sheet
 * Handles resource collection, consumption, military support, and build queue
 */
export class ResourcesPhase {
  private kingdomState: KingdomState;
  private turnManager: TurnManager;
  
  constructor(kingdomState: KingdomState, turnManager: TurnManager) {
    this.kingdomState = kingdomState;
    this.turnManager = turnManager;
  }
  
  render(): string {
    const production = this.kingdomState.calculateProduction();
    
    // Include resource styles
    const styles = `<style>${ResourceStyles.getStyles()}</style>`;
    
    return html`
      ${styles}
      <div class="resources-phase-container">
        <!-- Part 1: Resource Display at the Top -->
        ${this.renderResourceDisplay()}
        
        <!-- Part 2: Resource Steps Underneath -->
        ${this.renderResourceSteps()}
      </div>
    `;
  }
  
  /**
   * Part 1: Render the resource display at the top
   */
  private renderResourceDisplay(): string {
    return this.renderResourceDashboard();
  }
  
  /**
   * Part 2: Render all resource steps in a scrollable container
   */
  private renderResourceSteps(): string {
    return html`
      <div class="resource-steps-scroll">
        ${this.renderStep1CollectResources()}
        ${this.renderStep2FoodConsumption()}
        ${this.renderStep3MilitarySupport()}
        ${this.renderStep4BuildQueue()}
      </div>
    `;
  }
  
  /**
   * Render the resource dashboard (non-sticky)
   */
  private renderResourceDashboard(): string {
    return html`
      <div class="resource-dashboard-wrapper">
        <div class="resource-dashboard">
          ${this.renderCompactResourceCard('food', this.kingdomState.resources.get('food') || 0)}
          ${this.renderCompactResourceCard('lumber', this.kingdomState.resources.get('lumber') || 0)}
          ${this.renderCompactResourceCard('stone', this.kingdomState.resources.get('stone') || 0)}
          ${this.renderCompactResourceCard('ore', this.kingdomState.resources.get('ore') || 0)}
          ${this.renderCompactResourceCard('gold', this.kingdomState.resources.get('gold') || 0)}
        </div>
      </div>
    `;
  }
  
  /**
   * Render a compact resource card for the sticky dashboard
   */
  private renderCompactResourceCard(resource: string, value: number): string {
    let icon = '';
    let color = '';
    
    switch (resource) {
      case 'food': icon = 'fa-wheat-awn'; color = '#8B4513'; break;
      case 'lumber': icon = 'fa-tree'; color = '#228B22'; break;
      case 'stone': icon = 'fa-cube'; color = '#708090'; break;
      case 'ore': icon = 'fa-mountain'; color = '#4B0082'; break;
      case 'gold': icon = 'fa-coins'; color = '#FFD700'; break;
      default: icon = 'fa-question'; color = '#666666';
    }
    
    return html`
      <div class="resource-card">
        <i class="fas ${icon} resource-icon" style="color: ${color};"></i>
        <div class="resource-value">${value}</div>
        <div class="resource-label">${resource.charAt(0).toUpperCase() + resource.slice(1)}</div>
      </div>
    `;
  }
  
  /**
   * Step 1: Collect Resources
   */
  private renderStep1CollectResources(): string {
    const isCompleted = this.kingdomState.isPhaseStepCompleted('resources_collect');
    const production = this.kingdomState.calculateProduction();
    const hasWorksites = this.kingdomState.hexes.some(hex => hex.worksite !== null);
    
    let productionSummary = '';
    if (production.size > 0 || hasWorksites) {
      const productionText = production.size > 0 
        ? Array.from(production.entries())
            .map(([key, value]) => `+${value} ${key.charAt(0).toUpperCase() + key.slice(1)}`)
            .join(' | ')
        : 'No Production';
      
      let worksiteDetails = '';
      if (hasWorksites) {
        const worksiteList = this.kingdomState.hexes
          .filter(hex => hex.worksite !== null)
          .map(hex => {
            const prod = hex.getProduction();
            const prodText = prod.size > 0
              ? Array.from(prod.entries())
                  .map(([k, v]) => `${v} ${k.charAt(0).toUpperCase() + k.slice(1)}`)
                  .join(', ')
              : 'No production';
            return html`
              <li class="worksite-item">
                <span>${hex.name || `Hex ${hex.id}`} (${hex.terrain})</span>
                <span>${prodText}</span>
              </li>
            `;
          }).join('');
        
        worksiteDetails = html`
          <details style="margin-top: 10px;">
            <summary style="cursor: pointer; color: #666; font-size: 14px;">View Worksite Details</summary>
            <ul class="worksite-list" style="margin-top: 10px;">
              ${worksiteList}
            </ul>
          </details>
        `;
      }
      
      productionSummary = html`
        <div class="production-summary" style="margin: 15px 0; padding: 15px; background: linear-gradient(135deg, #f5f5f5 0%, #e5e5e5 100%); border-radius: 8px; border: 1px solid #d1d1d1;">
          <div style="display: flex; justify-content: space-between; align-items: center; margin-bottom: 10px;">
            <div class="production-summary-title" style="font-weight: bold; font-size: 16px;">Expected Income This Turn:</div>
            <div style="font-weight: bold; color: #22c55e;">${productionText}</div>
          </div>
          ${worksiteDetails}
        </div>
      `;
    } else {
      productionSummary = html`
        <div style="margin: 15px 0; padding: 15px; background: #f9f9f9; border-radius: 8px; border: 1px solid #e5e5e5; text-align: center; color: #666; font-style: italic;">
          No worksites currently producing resources
        </div>
      `;
    }
    
    const button = isCompleted
      ? html`
        <button class="turn-action-button" disabled style="opacity: 0.5;">
          <i class="fas fa-check"></i> Resources Collected
        </button>
      `
      : html`
        <button class="turn-action-button" onclick="window.executeResourceStep1()">
          <i class="fas fa-hand-holding-usd"></i> Collect Resources
        </button>
      `;
    
    return html`
      <div class="phase-step-container" style="position: relative;">
        ${isCompleted ? '<i class="fas fa-check-circle phase-step-complete"></i>' : ''}
        <strong>Step 1: Collect Resources and Revenue</strong>
        
        ${productionSummary}
        
        ${isCompleted ? '<div style="margin-top: 10px; color: #666; font-style: italic; text-align: center;">Resources have been collected for this turn.</div>' : ''}
        
        ${button}
      </div>
    `;
  }
  
  /**
   * Step 2: Food Consumption
   */
  private renderStep2FoodConsumption(): string {
    const isCompleted = this.kingdomState.isPhaseStepCompleted('resources_consumption');
    const totalFoodNeeded = this.kingdomState.getTotalFoodConsumption();
    const currentFood = this.kingdomState.resources.get('food') || 0;
    const shortage = Math.max(0, totalFoodNeeded - currentFood);
    
    const [settlementFood, armyFood] = this.kingdomState.getFoodConsumptionBreakdown();
    
    const warning = shortage > 0 && !isCompleted
      ? html`
        <div class="consumption-warning">
          <i class="fas fa-exclamation-triangle"></i>
          <strong>Warning:</strong> Food shortage of ${shortage} will cause +${shortage} Unrest!
        </div>
      `
      : '';
    
    const button = isCompleted
      ? html`
        <button class="turn-action-button" disabled style="opacity: 0.5;">
          <i class="fas fa-check"></i> Consumption Paid
        </button>
      `
      : html`
        <button class="turn-action-button" onclick="window.executeResourceStep2()">
          <i class="fas fa-utensils"></i> Pay Food Consumption
        </button>
      `;
    
    return html`
      <div class="phase-step-container" style="position: relative;">
        ${isCompleted ? '<i class="fas fa-check-circle phase-step-complete"></i>' : ''}
        <strong>Step 2: Food Consumption</strong>
        
        <div style="margin: 10px 0;">
          <div style="display: flex; justify-content: space-around; flex-wrap: wrap; gap: 20px;">
            <div style="text-align: center;">
              <i class="fas fa-home" style="font-size: 32px; color: #8B4513;"></i>
              <div style="font-size: 20px; font-weight: bold; margin: 2px 0;">${settlementFood}</div>
              <div style="font-size: 11px; color: #666;">Settlement Consumption</div>
            </div>
            
            <div style="text-align: center;">
              <i class="fas fa-shield-alt" style="font-size: 32px; color: #4682B4;"></i>
              <div style="font-size: 20px; font-weight: bold; margin: 2px 0;">${armyFood}</div>
              <div style="font-size: 11px; color: #666;">Army Consumption</div>
            </div>
            
            <div style="text-align: center;">
              <i class="fas fa-wheat-awn" style="font-size: 32px; color: ${shortage > 0 ? '#ef4444' : '#22c55e'};"></i>
              <div style="font-size: 20px; font-weight: bold; margin: 2px 0; color: ${shortage > 0 ? '#ef4444' : '#22c55e'};">
                ${currentFood} / ${totalFoodNeeded}
              </div>
              <div style="font-size: 11px; color: #666;">Available / Required</div>
            </div>
          </div>
          
          ${warning}
        </div>
        
        ${button}
      </div>
    `;
  }
  
  /**
   * Step 3: Military Support
   */
  private renderStep3MilitarySupport(): string {
    const isCompleted = this.kingdomState.isPhaseStepCompleted('resources_military');
    const totalSupport = this.kingdomState.getTotalArmySupport();
    const armyCount = this.kingdomState.armies.length;
    const unsupportedCount = Math.max(0, armyCount - totalSupport);
    
    let statusClass = 'good';
    if (unsupportedCount > 0) {
      statusClass = 'danger';
    } else if (armyCount === totalSupport) {
      statusClass = 'warning';
    }
    
    const unsupportedDisplay = unsupportedCount > 0
      ? html`
        <div class="support-status danger">
          <i class="fas fa-exclamation-triangle" style="font-size: 32px; color: #ef4444;"></i>
          <div>
            <div style="font-size: 18px; font-weight: bold; color: #ef4444;">${unsupportedCount}</div>
            <div style="font-size: 12px; color: #666;">Unsupported</div>
          </div>
        </div>
      `
      : '';
    
    const warning = unsupportedCount > 0 && !isCompleted
      ? html`
        <div class="consumption-warning">
          <i class="fas fa-exclamation-triangle"></i>
          <strong>Warning:</strong> ${unsupportedCount} unsupported ${unsupportedCount === 1 ? 'army' : 'armies'} will require morale checks!
        </div>
      `
      : armyCount === 0
      ? html`
        <div style="margin-top: 10px; color: #666; font-style: italic; text-align: center;">No armies currently fielded</div>
      `
      : '';
    
    const button = isCompleted
      ? html`
        <button class="turn-action-button" disabled style="opacity: 0.5;">
          <i class="fas fa-check"></i> Support Processed
        </button>
      `
      : html`
        <button class="turn-action-button" onclick="window.executeResourceStep3()">
          <i class="fas fa-flag"></i> Process Military Support
        </button>
      `;
    
    return html`
      <div class="phase-step-container" style="position: relative;">
        ${isCompleted ? '<i class="fas fa-check-circle phase-step-complete"></i>' : ''}
        <strong>Step 3: Military Support</strong>
        
        <div class="army-support-display">
          <div class="support-status ${statusClass}">
            <i class="fas fa-shield-alt army-icon"></i>
            <div>
              <div style="font-size: 18px; font-weight: bold;">${armyCount} / ${totalSupport}</div>
              <div style="font-size: 12px; color: #666;">Armies / Capacity</div>
            </div>
          </div>
          
          ${unsupportedDisplay}
        </div>
        
        ${warning}
        
        ${button}
      </div>
    `;
  }
  
  /**
   * Step 4: Build Queue
   */
  private renderStep4BuildQueue(): string {
    const isCompleted = this.kingdomState.isPhaseStepCompleted('resources_build');
    
    let buildQueueContent = '';
    if (this.kingdomState.buildQueue.length > 0) {
      const availableResources = ['lumber', 'stone', 'ore']
        .map(r => `${this.kingdomState.resources.get(r) || 0} ${r.charAt(0).toUpperCase() + r.slice(1)}`)
        .join(', ');
      
      const projects = this.kingdomState.buildQueue
        .map(project => this.renderBuildProject(project, isCompleted))
        .join('');
      
      buildQueueContent = html`
        <div class="build-queue-container">
          <div style="margin-bottom: 15px; font-weight: bold;">
            Available Resources: ${availableResources}
          </div>
          ${projects}
        </div>
      `;
    } else {
      buildQueueContent = html`
        <div style="margin-top: 10px; color: #666; font-style: italic; text-align: center;">No construction projects in queue</div>
      `;
    }
    
    let button = '';
    if (!isCompleted && this.kingdomState.buildQueue.length > 0) {
      button = html`
        <button class="turn-action-button" onclick="window.executeResourceStep4()">
          <i class="fas fa-hammer"></i> Apply to Construction
        </button>
      `;
    } else if (isCompleted) {
      button = html`
        <button class="turn-action-button" disabled style="opacity: 0.5;">
          <i class="fas fa-check"></i> Resources Applied
        </button>
      `;
    }
    
    return html`
      <div class="phase-step-container" style="position: relative;">
        ${isCompleted ? '<i class="fas fa-check-circle phase-step-complete"></i>' : ''}
        <strong>Step 4: Process Build Queue</strong>
        
        ${buildQueueContent}
        
        ${button}
      </div>
    `;
  }
  
  /**
   * Render a build project card
   */
  private renderBuildProject(project: BuildProject, isCompleted: boolean): string {
    const percentage = project.progress;
    const remaining: string[] = [];
    
    project.remainingCost.forEach((amount, resource) => {
      if (amount > 0) {
        remaining.push(`${amount} ${resource}`);
      }
    });
    
    const remainingText = remaining.length > 0 
      ? `Needs: ${remaining.join(', ')}`
      : 'Complete!';
    
    let allocationControls = '';
    if (!isCompleted && remaining.length > 0) {
      const controls: string[] = [];
      project.remainingCost.forEach((_, resource) => {
        const available = this.kingdomState.resources.get(resource) || 0;
        const icon = this.getResourceIcon(resource);
        const color = this.getResourceColor(resource);
        controls.push(html`
          <div class="allocation-control">
            <i class="fas ${icon}" style="color: ${color};"></i>
            <button class="allocation-button" onclick="window.adjustAllocation('${project.structureId}', '${resource}', -1)">−</button>
            <span class="allocation-value">0</span>
            <button class="allocation-button" onclick="window.adjustAllocation('${project.structureId}', '${resource}', 1)">+</button>
          </div>
        `);
      });
      
      allocationControls = html`
        <div class="resource-allocation">
          ${controls.join('')}
        </div>
      `;
    }
    
    return html`
      <div class="build-project-card">
        <div class="build-project-header">
          <span class="build-project-name">${project.settlementName}</span>
          <span class="build-project-tier">Building</span>
        </div>
        
        <div class="progress-bar-container">
          <div class="progress-bar-fill" style="width: ${percentage}%;">
            <span class="progress-bar-text">${percentage}%</span>
          </div>
        </div>
        
        <div style="font-size: 14px; color: #666; margin-top: 5px;">
          ${remainingText}
        </div>
        
        ${allocationControls}
      </div>
    `;
  }
  
  private getResourceIcon(resource: string): string {
    switch (resource) {
      case 'lumber': return 'fa-tree';
      case 'stone': return 'fa-cube';
      case 'ore': return 'fa-mountain';
      default: return 'fa-question';
    }
  }
  
  private getResourceColor(resource: string): string {
    switch (resource) {
      case 'lumber': return '#228B22';
      case 'stone': return '#708090';
      case 'ore': return '#4B0082';
      default: return '#666666';
    }
  }
}
